primary/tests/integration_tests.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
Is there also a way to unit test the implementation as well?

Reply:
Yes, I will work on this and ping this comment when its done.